### DISCUSSION TITLE
eglot with pyright reports errors on random non-existing variables in buffer

### DISCUSSION THREAD

**Discussion by x0xntor33alx0x** [joaotavora/eglot#1248] [General]
when running `eglot` on a python in a python project, sometimes the server complains about random non-existing variables not being defined, here is a showcase of the problem:
![](https://file.io/PhKT0z37zFXQ)

**steps to reproduce:**
1. start emacs 
2. open a python file
3. run `eglot`
4. after a while working in the buffer typing random text then deleting the server starts reporting a non-existing variables not being defined as above

I cannot reproduce the issue using emacs -Q. 

transcript: https://gist.github.com/x0xntor33alx0x/e5f6ec874c3a36448f6fa74445a7e8ff

**software versions**
emacs: GNU Emacs 29.0.92 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.20, cairo version 1.16.0) of 2023-06-25  **also tested on 28.0.50**
eglot: 1.12.29 (builtin) 
pyright: 1.1.315. **also tested with version 1.1.166**
OS: ubuntu 20.04

*PS: I've been using pyright with `lsp-mode` but never had this problem*

**Comment by joaotavora** [comment]
> I cannot reproduce the issue using emacs -Q.

So this means that your step _"1. start emacs"_ is actually _"1. start emacs and load all my configuration"_, right?

**Reply by x0xntor33alx0x** [reply]
yes, it works normally without any configurations.

**Reply by joaotavora** [reply]
> yes, it works normally without any configurations, so I guess the problem is in eglot itself.

How do you reach that conclusion?  If `emacs -Q` +  Eglot "works normally without any configurations", how can the problem be in Eglot?

**Reply by x0xntor33alx0x** [reply]
sorry I meant in the configuration, I just removed that.

**Reply by x0xntor33alx0x** [reply]
I was wondering if there is a fix for that, I tried disabling some of the modes in the buffer but the problem persists.

**Reply by joaotavora** [reply]
I think the next step is for you to follow the 7 step quige guide @ https://joaotavora.github.io/eglot/#Troubleshooting-Eglot closely.

You already have some of the elements, it seems.  Pay special attention to Emacs  so now you just have to package it altogether.  Pay special attention to step number 7, i.e. following your own recipe from scratch.
